### PR TITLE
Eq 389 Adding dynamic theme capacity

### PR DESCRIPTION
**Pull request by dhilton**
### What is the context of this PR?

This PR adds the ability for a survey schema to define a theme and for that theme (if it exists) to be used to display that survey in the runner.

~~We create a new theme 'titchfield' which is a stub for a social style theme.~~

A new Star Wars theme is added.
### How to review
1. Check out this branch.
2. Install new dependency `Flask-themes2`, by running `pip install -r requirements.txt`
3. Check that the tests pass and are sensible in their checks.
4. Run the star wars dev survey 
5. Notice the change of style and structure of the page.
- [x] Do all commits have sensible commit messages?
- [ ] Is all new code unit tested?
- [ ] Are there integration tests if needed?
- [ ] Is there sufficient logging?
- [ ] Is there documentation or is the code self-describing?
### Who worked on the PR

@dhilton, @LJBabbage and @hamishtaplin all worked on this PR in places. Thus none of those people should have the final say in merging this PR.

### CODE REVIEW

File: app/main/views/questionnaire.py [hunk 1]
```diff
@@ -46,12 +46,18 @@ def survey(eq_id, collection_id, location):
...
-
-        return render_template(template,
-                               meta=context['meta'],
-                               content=context['content'],
-                               navigation=context['navigation']
-                               )
+        try:
+            theme = context['meta']['survey']['theme']
+            logger.info("Theme selected: {} ".format(theme))
+        except KeyError:
```
Comment (LJBabbage) [inline app/main/views/questionnaire.py:52]: The scheme parser would fall over before here if theme was missing, so this might be redundant

**Comment by LJBabbage** [comment]
I get In 
--require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    simplejson from https://pypi.python.org/packages/79/9f/f6e2793254c8f87919d92abc42ef7d05aba5ca06ab590582ad32fa47be4f/simplejson-3.8.2-cp35-cp35m-macosx_10_9_x86_64.whl#md5=5a8d46930c05a45e3b6b35d5ff10478f (from Flask-Themes2==0.1.4-

**Comment by dhilton** [comment]
Good spot - I thought I had pinned those requirements, will hash them now.

**Comment by LJBabbage** [comment]
Looks good to me, I like that the new polite template which says hello when you are using it

**Comment by hamishtaplin** [comment]
Do you want to get this merged and I do a separate PR for the CSS?

**Comment by dhilton** [comment]
@hamishtaplin Don't mind - perhaps better to have it all in one?
@LJBabbage - Sorry should have removed that - is that still in the commits?

I'll rebase once everyone is happy with this.

**Comment by LJBabbage** [comment]
The Hello is at the bottom of _survey.html

**Comment by hamishtaplin** [comment]
Don't merge this yet as I'm working on it.

**Comment by hamishtaplin** [comment]
@this is gtg from my end but is failing tests

**Comment by dhilton** [comment]
This branch has been multi merged from master - rebasing is an absolute pain. I'm going to restage the commits and create a clean commit history for this feature as I can't seem to get a clean revert to work correctly without losing work.

File: app/model/questionnaire.py [hunk 2]
```diff
@@ -52,6 +53,78 @@ def _register_alias(self, item):
             elif self.aliases[item.alias] != item.id:
                 raise QuestionnaireException('{} is not a unique alias'.format(item.alias))
 
+    def to_json(self):
```
Comment (warrenbailey) [inline app/model/questionnaire.py]: This is the python model stuff. I don't think it should be here?

**Comment by warrenbailey** [comment]
All good apart from you've accidentally reintroduced the python model changes into app/model/questionnaire.py. These need to be removed 👎

**Comment by dhilton** [comment]
This should be good to go now @warrenbailey - apologies for including that depreciated code.

**Comment by warrenbailey** [comment]
LGTM 👍

**Pull request merged**